app/webpack/components/user_profile/profile.vue

```diff
@@ -0,0 +1,288 @@
+<template>
+  <div id="user-profile">
+    <div v-if="loading" class="text-center m-4">
+      <div class="spinner-border" role="status">
+        <span class="sr-only">Loading...</span>
+      </div>
+    </div>
+    <div v-if="showSignIn" id="user-authenticate">
+      <div v-if="showSignIn" class="alert alert-dismissible alert-warning">
+        <button
+          type="button"
+          class="close"
+          data-dismiss="alert"
+          aria-hidden="true"
+        >
+          &times;
+        </button>
+        {{ authenticateMessage }}
+      </div>
+      <sign-in
```

Comment:
I'm not sure why we need to have the sign in component here? Given the controller only allows you to go to the profile page if you're logged in we could just assume you are logged in within this component? That would save a lot of the complexity of this component? Unless there's a part of this profile flow that explicitly depends on the user being signed out for some part of the flow?

Reply:
I couldnt agree with you more. However, Firebase mandates that a user be freshly logged in to be able to modify their profile (https://firebase.google.com/docs/auth/web/manage-users#re-authenticate_a_user). This is why I am immediately logging a user off after a change is made, so that every change is done with a freshly logged in user

Reply:
Let's take this discussion to Slack etc. - there's bits around logging out the user that don't seem right too.